components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/Constants.java

```diff
@@ -103,15 +120,21 @@ public enum ErrorMessages {
         ERROR_CODE_USERNAME_NOT_PROVIDED("60002",
                                          "Username not provided.",
                                          "Username is not provided in the registration request of flow id: %s"),
-        ERROR_CODE_UNDEFINED_FLOW_ID("60003",
+        ERROR_CODE_USERNAME_ALREADY_EXISTS("60003",
+                                           "Username already exists.",
+                                           "The provided username already exists in the tenant: %s"),
+        ERROR_CODE_UNDEFINED_FLOW_ID("60004",
                                            "Flow id is not defined.",
                                            "The flow id is not defined in the registration request."),
-        ERROR_CODE_INVALID_USERNAME("60004",
+        ERROR_CODE_INVALID_USERNAME("60005",
                                      "Invalid username.",
                                      "The given username: %s must be an email address."),
-        ERROR_CODE_REQUEST_PROCESSING_FAILURE("60005",
-                                              "Error while processing the registration request.",
-                                              "While processing the registration request for flow id %s, server " +
+        ERROR_CODE_REGISTRATION_FAILURE("60006",
+                                              "Registration failed.",
+                                              "%s"),
+        ERROR_CODE_REQUEST_PROCESSING_FAILURE("60007",
+                                                      "Error while processing the registration request.",
+                                                      "While processing the registration request for flow id %s, server " +
                                                       "returned an error, \"%s\".");
```

Comment:
Shall we remove this description and add the description from the place it gets thrown